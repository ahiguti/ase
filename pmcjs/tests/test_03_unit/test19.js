
/* tied hash + method */

Perl.Use("TieTest");
var fs = Perl.Functions["TieTest"];
var ms = Perl.Methods["TieTest"];
var x = ms.newht(0);
x['a'] = 5;
x['b'] = 6;
x['c'] = 7;
print("xb=" + x['b']);
delete x['c'];
for (var i in x) {
  print("k=" + i + " v=" + x[i]);
}
var r = x.foo(15);
print(r);

