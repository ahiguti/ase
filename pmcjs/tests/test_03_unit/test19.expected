foo invoked
xb=6
k=a v=5
k=b v=6
30
